### PR TITLE
href channel

### LINKED ISSUE #283: Hyperlinks?
We need a way to make Plot marks linkable (meaning a + xlink:href).

### PR DESCRIPTION
**Pull request by mbostock**
Adds a generic href channel that wraps the specified mark elements with an A element with the appropriate xlink:href attribute. A target option is also available (e.g., “_blank”). I didn’t implement it for grouped marks (lines and areas), but maybe I should. Fixes #283.

### CODE REVIEW

File: src/style.js [hunk 1]
```diff
@@ -141,6 +147,17 @@ export function applyDirectStyles(selection, mark) {
   applyAttr(selection, "opacity", mark.opacity);
 }
 
+function applyHref(selection, href, target) {
+  selection.each(function(i) {
+    if (href[i] != null) {
+      const a = document.createElementNS(namespaces.svg, "a");
+      a.setAttributeNS(namespaces.xlink, "href", href[i]);
+      if (target != null) a.setAttribute("target", target);
+      this.parentNode.insertBefore(a, this).appendChild(this);
```
Comment (Fil) [inline src/style.js:156]: makes we wish we could do selection.wrap("svg:a") (related: https://github.com/d3/d3-selection/pull/296)

**Comment by Fil** [comment]
wonder if we shouldn't have target: "_blank" by default?

**Comment by mbostock** [comment]
> wonder if we shouldn't have target: "_blank" by default?

I don’t think so, no. I do wonder whether it’s possible to set the equivalent of a `<base target>` for the SVG though, so we don’t have to repeat the target option everywhere? It’d be nice to set it on the base plot somehow… but it’s not a huge deal to set it on each link, too.

**Comment by mbostock** [comment]
Any other changes, @Fil? Or LGTY?

**Review by Fil — APPROVED** [review]
it's good, we'll just have to add tests

File: src/style.js (hunk 1 above)
Comment (Fil) [reply src/style.js]: selection.wrap: https://github.com/d3/d3-selection/pull/297

**Pull request merged**